cli/src/base/move_package_resolver.rs

```diff
@@ -0,0 +1,200 @@
+use std::{collections::HashMap, fs};
+
+use anyhow::{anyhow, bail, Context, Result};
+use reqwest::blocking::Client;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use toml_edit::easy::{self, value::Table, Value};
+use utils::{env::MOVE_HOME, movey_credential};
+
+#[derive(Deserialize, Debug)]
+pub struct Dependencies {
+    pub dependencies: MoveyDependencies,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct MoveyDependencies {
+    pub movey: MoveyResolver,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct MoveyResolver {
+    pub resolver: String,
+    pub packages: Option<HashMap<String, String>>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct MovePackageResolver {
+    deps_map: HashMap<String, String>,
+    deps_resolved: HashMap<String, MoveDependency>,
+}
+
+impl MovePackageResolver {
+    fn new() -> Self {
+        MovePackageResolver {
+            deps_map: HashMap::new(),
+            deps_resolved: HashMap::new(),
+        }
+    }
+
+    pub fn execute() -> Result<()> {
+        let move_toml_content = String::from_utf8(
+            fs::read("Move.toml").context("Not in Move package root directory")?,
+        )?;
+
+        let mut deps = MovePackageResolver::new();
+        deps.parse_deps_from_toml(&move_toml_content)?;
+        deps.resolve_movey_deps(None)?;
+        let lock_file_content = deps.generate_lock_toml_content();
+
+        fs::write("Move.lock", lock_file_content)
+            .map_err(|e| anyhow!("Cannot write to Move.lock file. {e}"))?;
+        Ok(())
+    }
+
+    fn parse_deps_from_toml(&mut self, toml_content: &str) -> Result<()> {
+        let toml = easy::from_str::<Dependencies>(toml_content)
+            .map_err(|e| anyhow!("Wrong Move.toml format for Movey dependencies. Error: {e}"))?;
+        if toml.dependencies.movey.resolver != "movey" {
+            bail!("The CLI only resolve Movey dependencies.")
+        };
+        if let Some(offchain_deps) = toml.dependencies.movey.packages {
+            for (key, value) in offchain_deps {
+                self.deps_map.insert(key, value);
+            }
+        };
+        Ok(())
+    }
+
+    fn resolve_movey_deps(&mut self, url: Option<&str>) -> Result<()> {
+        let url = match url {
```

Comment:
Why don't we just use the url from `movey_credential`? Is there a scenario where we would want to override this url? 🤔

Reply:
Besides, is this API public (ie, not requiring token)? If that is the case, can we just provide a hardcoded base url?